src/iuse_actor.cpp

```diff
@@ -2236,9 +2236,12 @@ bool repair_item_actor::handle_components( player &pl, const item &fix,
     // Go through all discovered repair items and see if we have any of them available
     for( const auto &entry : valid_entries ) {
         const auto component_id = entry.obj().repaired_with();
-        if( crafting_inv.has_amount( component_id, items_needed ) ) {
-            // We've found enough of a material, add it to list
-            comps.push_back( item_comp( component_id, items_needed ) );
+        if( item::find_type( component_id )->count_by_charges() ) {
```

Comment:
Please use the static `item::count_by_charges` function, it takes the item type id as parameter.